### ISSUE TITLE
Feature: return log-probabilities from generate()

### ISSUE THREAD

**Issue by thiagosu** [fufuchiu/speechloom#27]
The `generate` function in `src/speechloom/model.py` currently returns
`list[int]` -- just the sampled token IDs. For research workflows (reranking,
perplexity computation, confidence-based early stopping) it would be very
useful to also get the log-probability of each generated token.

## Proposed API

```python
def generate(
    model: SpeechModel,
    audio: torch.Tensor,
    lengths: torch.Tensor,
    prompt_ids,
    max_new_tokens: int = 64,
    mode: str = 'mixed',
    return_logprobs: bool = False,
) -> list[int] | tuple[list[int], list[float]]:
```

When `return_logprobs=True`, the function returns a tuple of `(token_ids,
log_probs)` where `log_probs[i]` is the log-softmax value of the chosen token
at step `i`.

This is a backward-compatible additive change. The greedy path just needs to
capture `log_softmax(logits)[token]` before appending.

## Use case

I want to score multiple generated hypotheses and pick the one with the
highest mean log-probability, similar to length-normalized beam search scoring.
Without log-probs I have to re-run the model forward pass over the generated
sequence, which is wasteful.
